torch/fx/graph.py

```diff
@@ -358,6 +358,9 @@ def __deepcopy__(self, memo=None) -> 'Graph':
         g = Graph()
         output_val = g.graph_copy(self, val_map=memo)
         g.output(output_val)
+        for n, m in zip(g.nodes, self.nodes):
+            if n.op == 'output' and hasattr(m, 'type'):
+                n.type = m.type
```

Comment:
How about instead we do this:
1) Make `Graph.graph_copy` have a flag `return_output_node=False` that, when `True`, makes it so that the method returns a tuple of (new output value, old output node)
2) Call `graph_copy` with `return_output_node=True` at line 359 here
3) Pass the type from the original output node to the `output` call at line 360